custom_components/balboa/climate.py

```diff
@@ -65,17 +46,21 @@ def supported_features(self):
     @property
     def hvac_modes(self) -> List[str]:
         """Return the list of supported HVAC modes."""
-        return CLIMATE_SUPPORTED_MODES
+        if self._client.get_heatmode() == self._client.HEATMODE_RNR:
+            return [*CLIMATE_SUPPORTED_MODES, HVAC_MODE_AUTO]
+        else:
+            return CLIMATE_SUPPORTED_MODES
```

Comment:
This appears to change the available modes based on the current mode.  That doesn't seem right to me?

Reply:
@garbled1, so I've gone back and forth on this one and there are pros and cons to whichever way is decided. My reasoning for eventually going this way is because with the spa, you can only trigger setting the spa to "Ready" or "Rest" modes. You can not set the spa to "Ready in Rest" directly, but it can already be in that particular status mode. To trigger a spa to go to "Ready in Rest" mode, you have to toggle a pump while it is in "Rest" mode. Otherwise, you would have to set the spa to "Rest" mode first and then toggle a pump.

So if your spa is in "Rest"/"Off" or "Ready"/"Heat" mode and you list "Ready in Rest"/"Auto" as an available option, the user may be inclined to select the option. From there, we can do one of two things: set the mode to "Ready in Rest" which does nothing and will immediately report back as being in the previous mode, or set the mode to "Rest" and then toggle a pump so that it moves to the "Ready in Rest" status.

Neither of those options felt intuitive or desirable to me. Instead, I ended up removing "Ready in Rest"/"Auto" as an option for selecting unless the spa is already in that mode. That way, the mode is reflected correctly and the user knows they can only select the modes that are available to be triggered.

Happy to hear your thoughts on it!

Reply:
So I'm not sure offhand how often that function gets called.  I understand your reasoning..  I think my question is what is the new behavior?

If the spa goes into RnR, does the display update to say "auto" within a reasonable timeframe?  Or if you boot hass and the spa is in a non-RnR mode, does it just never show auto again?

Reply:
Ahh, ok! The new functionality is that if the spa is in RnR, it will show Auto/RnR. It updates right away, or at least I haven't seen a delay. Then when you switch to non-RnR, it disappears from the list immediately until the spa enters back into RnR mode, in which case it updates immediately as well.